jira/client.py

```diff
@@ -374,6 +377,7 @@ def __init__(self,
                  proxies=None,
                  timeout=None,
                  auth=None,
+                 generator=False,
```

Comment:
Please make ResultList act as a generator instead of adding a new parameter and different return values.